release_notes/v2.4.1.md

```diff
@@ -0,0 +1,144 @@
+v2.4.1 Release Notes - December 17, 2021
+========================================
+
+
+Improvements
+------------
+
+**peer - Chaincode as a service builder delivered with Fabric release artifacts and in fabric-peer docker image**
+
+Starting in v2.0 chaincode can be run as a service by utilizing the external builder pattern for chaincodes.
+Running 'chaincode as a service' has advantages in Kubernetes and other deployment environments since the chaincode can be managed independently rather than requiring the peer to build chaincode images and start chaincode containers at runtime.
+The external builder pattern required you to deliver a builder script or program alongside the peer.
+Starting in v2.4.1 an external builder for 'chaincode as a service' is available in the Fabric release artifacts and is pre-configured with the fabric-peer docker image,
+removing the need to build your own external builder and repackage and configure the peer.
+
+**peer - Gateway service now randomizes endorsement layouts**
+
+For improved load balancing of endorsement requests and fairer workload distribution across organizations,
+the gateway service now randomizes the peers it connects to for every transaction.
+
+**peer - Gateway service concurrency limits**
+
+Add support for setting gateway service concurrency limits, similar to how concurrency limits are
+set for the existing endorser and deliver services.
+Concurrency limits ensure that a peer does not process more than the configured number of requests,
+reducing the chance of a peer getting overloaded with more requests than it can process.
+The peer's core.yaml 'peer.limits.concurrency.gatewayService' setting defaults to 500.
+
+Fixes
+-----
+
+**peer - Fix for orderer endpoint updates in channel configuration transactions**
+
+When processing a channel configuration transaction, the gateway service was setting orderer endpoints based on the
+prior configuration and therefore the gateway may attempt to submit transactions to outdated orderer endpoints.
+A peer restart was required to refresh the orderer endpoints.
+The gateway now sets the orderer endpoints based on the updated channel configuration without a peer restart required.
+
+**peer - Extraneous warnings logged for service discovery endorsers requests**
+
+The 'discovery.acl' logger was logging an unexpected "identity does not satisfy principal" warning message.
+The message has now been removed.
+
+**peer - Improve gateway service memory garbage collection**
+
+Remove duplicate proposal responses from gateway service memory to improve garbage collection and peer process memory footprint.
+
+**peer - Reduce the size of gateway service gRPC error responses for non-matching endorsements**
+
+The gateway service was returning error responses with full payloads to clients when it received non-matching endorsements from peers.
```

Comment:
Another instance implying a prior published version of gateway service. (i.e. "was returning").